### PR TITLE
La 264 kpi events redesign

### PR DESCRIPTION
**Pull request by Vadym-Lopatka**

### CODE REVIEW

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **vadym-lopatka**
You can assign the PR to them by writing `/assign @vadym-lopatka` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
@Vadym-Lopatka: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | a1a7a4b5b13d51dbd94bc73de11d25f31d43c844 | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-128-pr-txnws) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by selina-bot-user** [comment]
@Vadym-Lopatka: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 0c6c982035d5c95e2e6960d7a7bb30dc15bd5756 | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-128-pr-rvpf5) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-128** [here](http://ms-lending-api-pr128.selinadev.co.uk)

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-128** [here](http://ms-lending-api-pr128.selinadev.co.uk)

File: src/main/java/com/selina/lending/messaging/event/BrokerRequestKpiEvent.java [hunk 1]
```diff
@@ -23,21 +23,18 @@
...
         String httpResponseBody,
         Integer httpResponseCode,
-        String decision,
         Instant started,
         Instant finished
 ) implements KafkaEvent {
     @Override
     public String key() {
-        return requestId;
+        return source;
```
[the remark this answers is not in the document]
Comment (Vadym-Lopatka) [reply src/main/java/com/selina/lending/messaging/event/BrokerRequestKpiEvent.java:38]: Yes.
The key is using for partitioning.
For partitioning purpose it mustn't be uniq but will guaranty specific order where all events of the specific broker will be ordered.
In my opinion, it is what we need here.

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[bpoon-selina](https://github.com/Selina-Finance/ms-lending-api/pull/128#issuecomment-1428171742 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)~~ [bpoon-selina]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**